python/beeai_framework/agents/experimental/__init__.py

```diff
@@ -1,7 +1,22 @@
 # Copyright 2025 © BeeAI a Series of LF Projects, LLC
 # SPDX-License-Identifier: Apache-2.0
 
+from typing import Any
+
 from beeai_framework.agents.experimental.agent import RequirementAgent
-from beeai_framework.agents.experimental.types import RequirementAgentRunOutput, RequirementAgentRunState
+from beeai_framework.agents.experimental.types import (
+    RequirementAgentOutput,
+    RequirementAgentRunState,
+)
+
+
+# lazy import deprecated aliases to prevent unecessary warnings
+def __getattr__(name: str) -> Any:
```

Comment:
This is all extremely verbose and is all over the place. Consider removing all deprecating fallbacks and including a migration guide in the PR instead so that we can announce it as part of the release.

Reply:
Should we keep the export of the deprecated aliases in the init modules? The idea of adding the fallbacks is so that we could emit warnings on them in case they get imported by consumers/users of the framework.